purchase_landed_cost/models/purchase_cost_distribution.py

```diff
@@ -562,12 +562,26 @@ def _compute_display_name(self):
     @api.onchange('type')
     def onchange_type(self):
         if self.type and self.type.default_amount:
-            self.expense_amount = self.type.default_amount
+            # set expense_amount in the currency of the distribution
+            currency_from = self.type.company_id.currency_id
+            amount = self.type.default_amount
+            currency_to = self.distribution.currency_id
+            company = self.company_id or self.env.user.company_id
+            cost_date = self.distribution.date or fields.Date.today()
+            self.expense_amount = currency_from._convert(amount, currency_to,
+                                                         company, cost_date)
 
     @api.onchange('invoice_line')
     def onchange_invoice_line(self):
         self.invoice_id = self.invoice_line.invoice_id.id
-        self.expense_amount = self.invoice_line.price_subtotal
+        # set expense_amount in the currency of the distribution
```

Comment:
Good text self explanatory but might be better as the docstring of the function?